### PR TITLE
docs: add CORS policy related documentation

### LINKED ISSUE #2943: add CORS documentation to HTTPProxy reference
We need to add documentation on CORS support to the HTTPProxy reference (https://github.com/projectcontour/contour/blob/main/site/docs/main/httpproxy.md).

It probably makes sense to add a section under https://github.com/projectcontour/contour/blob/main/site/docs/main/httpproxy.md#virtual-host-configuration, similar to the existing section there on TLS.

Followup from #2890

### PR DESCRIPTION
**Pull request by aberasarte**
Closes #2943

### CODE REVIEW

**Comment by aberasarte** [comment]
I added a CORS policy block to the HTTPProxy reference with some basic information. I'm not sure whether a more thorough explanation of CORS and the involved headers is needed here.

File: site/docs/main/httpproxy.md [hunk 1]
```diff
@@ -403,6 +403,74 @@ spec:
...
+      maxAge: "10m"
+  routes:
+    - conditions:
+      - prefix: /
+      services:
+        - name: s1
+          port: 80
+```
+
+`MaxAge` durations are expressed in the Go [duration format](https://godoc.org/time#ParseDuration). Valid time units are "ns", "us" (or "µs"), "ms", "s", "m", "h". Only positive values are allowed and 0 disables the cache requiring a preflight OPTIONS	check for all cross-origin requests.
```
Comment (skriss) [inline site/docs/main/httpproxy.md:472]: nit: there's a tab between "OPTIONS" and "check"

**Review by skriss — APPROVED** [review]
Thanks for writing this up @aberasarte! It looks reasonable to me, just one minor nit.

File: site/docs/main/httpproxy.md [hunk 2]
```diff
@@ -403,6 +403,74 @@ spec:
...
+
+In the following example, cross-domain requests are restricted to https://client.example.com only.
+
+```yaml
+apiVersion: projectcontour.io/v1
+kind: HTTPProxy
+spec:
+  virtualhost:
+    fqdn: www.example.com
+  corsPolicy:
```
Comment (stevesloka) [inline site/docs/main/httpproxy.md:449]: This should be indented one more to be inside the `virtualhost` struct.

File: site/docs/main/httpproxy.md [hunk 3]
```diff
@@ -403,6 +403,74 @@ spec:
...
+
+Contour allows configuring the headers involved in cross-domain requests. In this example, cross-domain requests will be allowed for any domain (note the `*` value).
+
+```yaml
+apiVersion: projectcontour.io/v1
+kind: HTTPProxy
+spec:
+  virtualhost:
+    fqdn: www.example.com
+  corsPolicy:
```
Comment (stevesloka) [inline site/docs/main/httpproxy.md:418]: This should be indented one more to be inside the `virtualhost` struct.

**Review by stevesloka — CHANGES_REQUESTED** [review]
Just two small nits in the example yaml files.

File: site/docs/main/httpproxy.md (hunk 2 above)
Comment (skriss) [reply site/docs/main/httpproxy.md:449]: good catch!

**Comment by stevesloka** [comment]
Also @aberasarte you need to sign the commit to make the DCO robots happy. =)

**Review by skriss — APPROVED** [review]
LGTM, thanks for making the updates @aberasarte.

**Pull request merged**